lib/util/segments.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+/**
+ * Increments the counts on transaction and agent
+ * around total segments
+ *
+ * @param {Transaction} transaction active transaction
+ */
+module.exports = function incrementSegments(transaction) {
```

Comment:
Can this be a method on `Transaction`? Maybe `transaction.incrementSegmentCounts()`?

Regardless, it's not clear to me from the doc block why/when we would need to do this.